executor/tiflashtest/tiflash_test.go

```diff
@@ -1452,3 +1452,127 @@ func TestMPPMemoryTracker(t *testing.T) {
 	require.NotNil(t, err)
 	require.True(t, strings.Contains(err.Error(), "Out Of Memory Quota!"))
 }
+
+func TestDisaggregatedTiFlashGeneratedColumn(t *testing.T) {
+	store := testkit.CreateMockStore(t, withMockTiFlash(2))
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+
+	tk.MustExec("drop table if exists t1;")
+	tk.MustExec("create table t1(c1 varchar(100), c2 varchar(100) AS (lower(c1)));")
+	tk.MustExec("insert into t1(c1) values('ABC');")
+	tk.MustExec("alter table t1 set tiflash replica 1;")
+	tb := external.GetTableByName(t, tk, "test", "t1")
+	err := domain.GetDomain(tk.Session()).DDL().UpdateTableReplicaInfo(tk.Session(), tb.Meta().ID, true)
+	require.NoError(t, err)
+
+	tk.MustExec("drop table if exists t2;")
+	tk.MustExec("create table t2(c1 int, c2 varchar(100));")
+	tk.MustExec("insert into t2 values(1, 'xhy'), (2, 'abc');")
+	tk.MustExec("alter table t2 set tiflash replica 1;")
+	tb = external.GetTableByName(t, tk, "test", "t2")
+	err = domain.GetDomain(tk.Session()).DDL().UpdateTableReplicaInfo(tk.Session(), tb.Meta().ID, true)
+	require.NoError(t, err)
+	tk.MustExec("alter table t2 add index idx2((lower(c2)));")
+
+	nthPlan := 100
+	test1 := func(forceTiFlash bool) {
+		if forceTiFlash {
+			tk.MustExec("set tidb_isolation_read_engines = 'tiflash'")
+		} else {
+			tk.MustExec("set tidb_isolation_read_engines = 'tikv,tiflash'")
+		}
+		sqls := []string{
+			"explain select /*+ nth_plan(%d) */ * from t2 where lower(c2) = 'abc';",
+			"explain select /*+ nth_plan(%d) */ count(*) from t2 where lower(c2) = 'abc';",
+			"explain select /*+ nth_plan(%d) */ count(c1) from t2 where lower(c2) = 'abc';",
+		}
+		for _, sql := range sqls {
+			var genTiFlashPlan bool
+			var selectionPushdownTiFlash bool
+			var aggPushdownTiFlash bool
+
+			for i := 0; i < nthPlan; i++ {
+				s := fmt.Sprintf(sql, i)
+				fmt.Printf("handling sql: %s, forceTiFlash: %v\n", s, forceTiFlash)
+				rows := tk.MustQuery(s).Rows()
+				for _, row := range rows {
+					line := fmt.Sprintf("%v", row)
+					if strings.Contains(line, "tiflash") {
```

Comment:
maybe distinguish the `mpp` keyword is more effective

Reply:
ditto for the below

Reply:
In non-disaggregated tiflash mod, may generate batchCop/cop plan instead of mpp plan.